src/compiler/emitter.ts

```diff
@@ -1191,6 +1195,10 @@ namespace ts {
             nodeIdToGeneratedName = [];
             autoGeneratedIdToGeneratedName = [];
             generatedNames = new Set();
+            formattedNameTempFlagsStack = [];
+            formattedNameTempFlags = new Map();
+            privateNameTempFlagsStack = [];
+            privateNameTempFlags = TempFlags.Auto;

```

Comment:
Pardon?

Reply:
sorry for the slang. I just meant that we have to write this save/restore pattern everywhere we want to emulate dynamic scope, and it would be nice if Javascript had dynamically scoped variables. They are so rarely used and easily abused that it's a not realistic language feature, though.